fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/rescheduleloan/data/LoanRescheduleRequestDataValidator.java

```diff
@@ -140,11 +141,85 @@ public void validateForCreateAction(final JsonCommand jsonCommand, final Loan lo
             final LocalDate endDate = jsonCommand.localDateValueOfParameterNamed(RescheduleLoansApiConstants.endDateParamName);
             final BigDecimal emi = jsonCommand.bigDecimalValueOfParameterNamed(RescheduleLoansApiConstants.emiParamName);
+            validateForCumulativeLoan(dataValidatorBuilder, loan, jsonElement, rescheduleFromDate, endDate, emi);
+        } else {
+            validateForProgressiveLoan(dataValidatorBuilder, loan, jsonElement, rescheduleFromDate, adjustedDueDate);
+        }
+
+        if (!dataValidationErrors.isEmpty()) {
+            throw new PlatformApiDataValidationException(dataValidationErrors);
+        }
+    }
+
+    private void validateForProgressiveLoan(final DataValidatorBuilder dataValidatorBuilder, final Loan loan, final JsonElement jsonElement,
+            final LocalDate rescheduleFromDate, final LocalDate adjustedDueDate) {
+        final var unsupportedFields = List.of(RescheduleLoansApiConstants.graceOnPrincipalParamName, //
+                RescheduleLoansApiConstants.graceOnInterestParamName, //
+                RescheduleLoansApiConstants.extraTermsParamName, //
+                RescheduleLoansApiConstants.emiParamName//
+        );
+
+        for (var unsupportedField : unsupportedFields) {
+            if (this.fromJsonHelper.parameterHasValue(unsupportedField, jsonElement)) {
+                dataValidatorBuilder.reset().parameter(unsupportedField).failWithCode(
+                        RescheduleLoansApiConstants.rescheduleSelectedOperationNotSupportedErrorCode,
+                        "Selected operation is not supported at a time during Loan Rescheduling");
+                return;
+            }
+        }
+
+        final LocalDate businessDate = DateUtils.getBusinessLocalDate();
+        LoanRepaymentScheduleInstallment installment = null;
+        if (rescheduleFromDate != null) {
+            boolean hasInterestRateChange = this.fromJsonHelper.parameterHasValue(RescheduleLoansApiConstants.newInterestRateParamName,
+                    jsonElement);
+            if (hasInterestRateChange && adjustedDueDate != null) {
+                dataValidatorBuilder.reset().parameter(RescheduleLoansApiConstants.adjustedDueDateParamName).failWithCode(
+                        RescheduleLoansApiConstants.rescheduleMultipleOperationsNotSupportedErrorCode,
+                        "Only one operation is supported at a time during Loan Rescheduling");
+                return;
+            }
+
+            if (hasInterestRateChange && !rescheduleFromDate.isAfter(businessDate)) {
+                dataValidatorBuilder.reset().parameter(RescheduleLoansApiConstants.rescheduleFromDateParamName).failWithCode(
+                        "loan.reschedule.interestratechange.reschedulefrom.shouldstartnextday",
+                        "Loan Reschedule From date should start on the next business day.");
```

Comment:
Misleading error message. It must be future date, but does not necessarily need to be the next business day...

Reply:
yes could be, we can find a better way to handle it.
for me the next business date is that day which is specified by the organisation. If no holiday setup then every day is a business day. The first message that I used was exactly same "Should be in the future".

Reply:
I am not fully sure how holidays play a part here... Are you happy to change the error message or you wanna keep this one?

Reply:
No, I will change it. It should be more self-expressive to avoid misunderstanding.